api/seccompprofile/v1beta1/seccompprofile_types.go

```diff
@@ -183,3 +212,30 @@ type SeccompProfileList struct {
 func init() { //nolint:gochecknoinits // required to init scheme
 	SchemeBuilder.Register(&SeccompProfile{}, &SeccompProfileList{})
 }
+
+func UnionSyscalls(baseSyscalls, appliedSyscalls []*Syscall) []*Syscall {
+	allSyscalls := make(map[seccomp.Action]map[string]bool)
+	for _, b := range baseSyscalls {
+		allSyscalls[b.Action] = make(map[string]bool)
+		for _, n := range b.Names {
+			allSyscalls[b.Action][n] = true
+		}
+	}
+	for _, s := range appliedSyscalls {
+		if _, ok := allSyscalls[s.Action]; !ok {
+			allSyscalls[s.Action] = make(map[string]bool)
+		}
+		for _, n := range s.Names {
+			allSyscalls[s.Action][n] = true
+		}
+	}
+	finalSyscalls := make([]*Syscall, 0, len(appliedSyscalls)+len(baseSyscalls))
```

Comment:
doesn't this allocate more memory than you might need? some, if not most, of the entries of these profiles will be the same. Seems to me like it would be preferable to allocate only for the length of the longest array. The rest of the allocations would be handled via the `append` function.

Reply:
yes, probably

Reply:
Done in a separate commit.